paper-dropdown-menu.html

```diff
@@ -7,178 +8,299 @@
+            value: false,
+            reflectToAttribute: true
+        },
+
+        /**
+         * Set to true to always float the label. Bind this to the
+         * `<paper-input-container>`'s `alwaysFloatLabel` property.
+         */
+        alwaysFloatLabel: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * Set to true to disable animations when opening and closing the
+         * dropdown.
+         */
+        noAnimations: {
+          type: Boolean,
+          value: false
+        }
+      },
+
+      keyBindings: {
+        'up down': 'open',
+        'esc': 'close'
+      },
+
+      hostAttributes: {
+        role: 'group',
+        'aria-haspopup': 'true'
+      },
+
+      /**
+       * Show the dropdown content.
        */
-      openedIcon: 'arrow-drop-up',
+      open: function() {
+        this.$.menuButton.open();
+      },
 
       /**
-       * The icon to display when the drop-down is closed.
+       * Hide the dropdown content.
+       */
+      close: function() {
+        this.$.menuButton.close();
+      },
+
+      /**
+       * A handler that is called when `iron-activate` is fired.
        *
-       * @attribute closedIcon
-       * @type string
-       * @default 'arrow-drop-down'
+       * @param {CustomEvent} event An `iron-activate` event.
        */
-      closedIcon: 'arrow-drop-down'
-
-    },
-
-    selectedItemLabel: '',
-
-    overlayListeners: {
-      'core-overlay-open': 'openAction',
-      'core-activate': 'activateAction',
-      'core-select': 'selectAction'
-    },
-
-    activateAction: function(e) {
-      this.opened = false;
-    },
-
-    selectAction: function(e) {
-      var detail = e.detail;
-      if (detail.isSelected) {
-        this.$.label.classList.add('selectedItem');
-        this.selectedItemLabel = detail.item.label || detail.item.textContent;
-      } else {
-        this.$.label.classList.remove('selectedItem');
-        this.selectedItemLabel = '';
-      }
-    }
+      _onIronActivate: function(event) {
+        this._setSelectedItem(event.detail.item);
+      },
 
-  };
+      /**
+       * A handler that is called when the `paper-input` is tapped.
+       *
+       * @param {CustomEvent} event A tap event.
+       */
+      _onInputTap: function(event) {
+        if (!this.opened) {
+          this.$.menuButton.open();
+        }
+      },
 
-  Polymer.mixin2(p, Polymer.CoreFocusable);
-  Polymer(p);
+      /**
+       * Computed the label for the dropdown given a selected item.
+       *
+       * @param {Element} selectedItem A selected Element item, with an
+       * optional `label` property.
+       */
+      _computeSelectedItemLabel: function(selectedItem) {
+        if (!selectedItem) {
+          return '';
+        }
 
-})();
+        return selectedItem.label || selectedItem.textContent.trim();
+      },
 
+      _computeMenuVerticalOffset: function(noLabelFloat) {
+        return noLabelFloat ? -4 : 16;
```

Comment:
They are magical because they are based on the current style values for internal elements of `paper-input`. Do you know a good way for me to derive the metrics of the input? (the metrics change depending on whether or not it has a floating label, among other things)